### DISCUSSION TITLE
Turning off compartment if IPRED < 0

### DISCUSSION THREAD

**Discussion by JonKnights** [nlmixrdevelopment/nlmixr#603] [Q&A] [answered]
Hi!

In nlmixr, is there a way to specify a zero-order elimination pathway and turn the compartment off when the amount in the compartment is predicted to reach zero?

Something like:
d/dt(A) = -k
if(A<=0){k=0}

**Comment by mattfidler** [comment, marked as answer]
In theory this should work.  However, this comparison will not happen when `A` exactly turns zero.  If `A` is negative it will stay negative for the duration with something like this.

You could also use the `stateTrim` solving option to handle situations like this.

Here is the `rxSolve` option:

```
stateTrim: When amounts/concentrations in one of the states are above
          this value, trim them to be this value. By default Inf.  Also
          trims to -stateTrim for large negative
          amounts/concentrations.  If you want to trim between a range
          say ‘c(0, 2000000)’ you may specify 2 values with a lower and
          upper range to make sure all state values are in the
          reasonable range.
```

**Reply by JonKnights** [reply]
Thanks Matt!
